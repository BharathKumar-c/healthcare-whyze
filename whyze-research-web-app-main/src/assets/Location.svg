<svg width="24" height="20" viewBox="0 0 24 20" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16.2752 1.33594C15.4502 1.33594 14.6627 1.48594 13.9127 1.74844C12.4877 0.660937 10.7252 0.0234375 8.81269 0.0234375C4.20019 0.0234375 0.450195 3.77344 0.450195 8.38594C0.450195 12.8859 8.06269 19.5609 8.36269 19.8234C8.47519 19.9359 8.62519 19.9734 8.77519 19.9734C8.92519 19.9734 9.07519 19.9359 9.18769 19.8234C9.33769 19.6734 11.2127 18.0234 13.1252 15.8484C14.5127 17.3484 15.7502 18.4359 15.7877 18.4734C15.9002 18.5859 16.0502 18.6234 16.2002 18.6234C16.3502 18.6234 16.5002 18.5859 16.6127 18.4734C16.8752 18.2484 23.4377 12.4734 23.4377 8.57344C23.5502 4.59844 20.2877 1.33594 16.2752 1.33594ZM8.81269 18.4359C6.52519 16.3359 1.8002 11.3859 1.8002 8.38594C1.8002 4.52344 4.95019 1.33594 8.85019 1.33594C12.7502 1.33594 15.9002 4.48594 15.9002 8.38594C15.8627 11.3859 11.1377 16.3359 8.81269 18.4359ZM16.2752 17.1234C15.7877 16.6734 14.9252 15.8484 14.0252 14.8734C15.7127 12.7734 17.1752 10.3734 17.1752 8.38594C17.1752 6.24844 16.3502 4.29844 15.0377 2.79844C15.4502 2.72344 15.8627 2.64844 16.2752 2.64844C19.5377 2.64844 22.2002 5.31094 22.2002 8.57344C22.2377 11.1609 18.2627 15.2859 16.2752 17.1234Z" fill="#0D5959"/>
<path d="M8.81231 4.48438C6.7123 4.48438 4.9873 6.20938 4.9873 8.30938C4.9873 10.4094 6.7123 12.1344 8.81231 12.1344C10.9123 12.1344 12.6373 10.4094 12.6373 8.30938C12.6373 6.20938 10.9123 4.48438 8.81231 4.48438ZM8.81231 10.8219C7.4248 10.8219 6.2998 9.69688 6.2998 8.30938C6.2998 6.92188 7.4248 5.79688 8.81231 5.79688C10.1998 5.79688 11.3248 6.92188 11.3248 8.30938C11.3248 9.69688 10.1998 10.8219 8.81231 10.8219Z" fill="#0D5959"/>
</svg>
