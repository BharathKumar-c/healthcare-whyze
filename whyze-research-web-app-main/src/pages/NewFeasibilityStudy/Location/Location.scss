.Location-card {
    border-left: #E5718C;



}

.location-container {

    h3 {
        font-weight: 700;
    }

    .checkedtree {

        .ant-tree-node-content-wrapper {
            background-color: #D9F042 !important;
        }
    }

    &-country-tree {
        margin: 15px 0 0;
        width: 50%;
    }

    .ant-tree-treenode-switcher-open,
    .ant-tree-treenode-switcher-close {
        margin: 5px 0 0;
    }

    .ant-tree-treenode-checkbox-checked {
        width: 100%;
        background: #D9F042;
        margin: 5px 0 0;
    }

    .ant-tree-node-content-wrapper {
        margin: 5px 0 0;
        background-color: unset !important;
        border-radius: 2px;
    }

}

.ant-tree-checkbox-checked .ant-tree-checkbox-inner {
    background-color: #2D3339 !important;
    color: #D9F042 !important;
    border: none !important;
}

.ant-tree-checkbox-indeterminate .ant-tree-checkbox-inner:after {
    background-color: #2D3339 !important;
    color: #D9F042 !important;
    border: none !important;
    width: 14px !important;
    height: 14px !important;
}