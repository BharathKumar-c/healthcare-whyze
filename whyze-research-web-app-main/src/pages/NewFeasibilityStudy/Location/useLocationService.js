import { useDispatch, useSelector } from 'react-redux';
import { setLoading } from '../../../redux/reducer/authReducer';
import ApiUtil from '../../../utils/ApiUtils';
import { feasibilityStudyApiUrls } from '../../../constant/ApiEndPoints';
import { setSegments } from '../../../redux/reducer/feasibilityStudyReducer';

const useLocationService = () => {
  const dispatch = useDispatch();
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const newSegments = JSON.parse(JSON.stringify(segments));
  const getSitesList = async (data) => {
    dispatch(setLoading(true));

    await ApiUtil.postData(`${feasibilityStudyApiUrls.sitesDataUrl}`, data)
      .then((res) => {
        newSegments[activeSegment.index].sites = res?.data || [];

        newSegments[activeSegment.index].siteCount = {
          ...newSegments[activeSegment.index].siteCount,
          location: res?.data?.length || 0,
        };
        newSegments[activeSegment.index].patientCount = {
          ...newSegments[activeSegment.index].patientCount,
          location: newSegments[activeSegment.index].selectedCountries.reduce(
            (sum, { count }) => sum + count,
            0,
          ),
        };
        dispatch(setSegments(newSegments));
      })
      .catch((err) => {
        console.log(err);
      });

    dispatch(setLoading(false));
  };

  return {
    getSitesList,
  };
};

export default useLocationService;
