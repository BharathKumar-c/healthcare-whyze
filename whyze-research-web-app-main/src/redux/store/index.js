import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
