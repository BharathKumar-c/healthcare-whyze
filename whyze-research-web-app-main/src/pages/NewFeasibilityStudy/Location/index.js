import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import Treecomponenet from '../../../components/hoc/tree';
import './Location.scss';
import { Card } from '../../../components/hoc';
import NoOfPatient from '../../../components/hoc/card/NoOfPatient';
import { useSelector } from 'react-redux';
import { useDispatch } from 'react-redux';
import { setSegments } from '../../../redux/reducer/feasibilityStudyReducer';
import {
  getSelectedLocationKeys,
  locationHistoryTreeFormatter,
  updateSelectedCountries,
} from '../../../utils/Formatter';
import useLocationService from './useLocationService';

const Location = ({ location }) => {
  const { segments, activeSegment } = useSelector(
    (state) => state.feasibilityStudyReducer,
  );
  const { getSitesList } = useLocationService();
  const dispatch = useDispatch();

  const [locationTree, setLocationTree] = useState([]);
  const [expandedKeys, setExpandedKeys] = useState([]);
  const [checkedKeys, setCheckedKeys] = useState([]);
  const [topCountriesTree, setTopCountryTree] = useState([]);
  const [topCountriesCheckedKeys, setTopCountriesCheckedKeys] = useState([]);
  const [isGetSites, setIsGetSites] = useState(false);

  const topCountriesList =
    location &&
    [...location]
      ?.sort((a, b) => Number(b.count) - Number(a.count))
      .slice(0, 2);

  useEffect(() => {
    const formattedLocation = locationHistoryTreeFormatter(location, true);
    const formattedTopCountries = locationHistoryTreeFormatter(
      topCountriesList,
      true,
    );
    const seletcedLocationKeys = getSelectedLocationKeys({
      locations: formattedLocation[0]?.children || [],
      selectedCountries: segments[activeSegment.index].selectedCountries,
    });

    const seletcedTopCountriesKeys = getSelectedLocationKeys({
      locations: formattedTopCountries[0]?.children || [],
      selectedCountries: segments[activeSegment.index].selectedCountries,
    });

    setCheckedKeys(seletcedLocationKeys);
    setLocationTree(formattedLocation[0]?.children || []);
    setTopCountriesCheckedKeys(seletcedTopCountriesKeys);
    setTopCountryTree(formattedTopCountries[0]?.children || []);
  }, [location]);

  const onExpand = (expandedKeysValue) => {
    setExpandedKeys(expandedKeysValue);
  };

  const onCheck = (value, isCountry) => {
    const newSegment = JSON.parse(JSON.stringify(segments));
    newSegment[activeSegment.index].selectedCountries = updateSelectedCountries(
      {
        data: isCountry
          ? newSegment[activeSegment.index].location
          : topCountriesList,
        value: value.checked,
        key: value.node.key,
        selectedCountries:
          newSegment[activeSegment.index].selectedCountries || [],
      },
    );
    setIsGetSites(true);
    dispatch(setSegments(newSegment));
  };

  useEffect(() => {
    if (isGetSites) {
      getSitesList({
        caseStudy: segments[activeSegment.index].feasibilityStudy,
        country: segments[activeSegment.index].selectedCountries,
      });
      setIsGetSites(false);
    }
  }, [isGetSites]);

  useEffect(() => {
    if (!isGetSites) {
      getSitesList({
        caseStudy: segments[activeSegment.index].feasibilityStudy,
        country: segments[activeSegment.index].selectedCountries,
      });
    }
  }, [segments[activeSegment.index]?.selectedCountries?.length > 0]);

  return (
    <Card
      title={{ case: 'Location', name: 'Location' }}
      tagtitle={'Inclusion'}
      className="location-Card"
      extra={<NoOfPatient totalpatient={0} />}
    >
      <div className="location-container">
        <div className="location-container-country-tree">
          <h3>Top countries</h3>
          {topCountriesList.length > 0 && (
            <Treecomponenet
              treeData={topCountriesTree}
              checkable
              title={'Location'}
              onCheck={(_, event) => onCheck(event, false)}
              checkedKeys={topCountriesCheckedKeys}
              showLine={false}
            />
          )}
        </div>
        <div className="location-container-country-tree">
          <h3>Countries</h3>
          <Treecomponenet
            treeData={locationTree}
            checkable
            title={'Location'}
            onExpand={onExpand}
            expandedKeys={expandedKeys}
            onCheck={(_, event) => onCheck(event, true)}
            checkedKeys={checkedKeys}
            showLine={false}
          />
        </div>
        <div></div>
      </div>
    </Card>
  );
};

Location.propTypes = {
  location: PropTypes.array,
};
Location.defaultProps = {
  location: [],
};

export default Location;
