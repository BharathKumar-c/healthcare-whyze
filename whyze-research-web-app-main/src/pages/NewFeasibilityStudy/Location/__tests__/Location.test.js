import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import Location from '..';
import store from '../../../../redux/store';

function renderWithContext(element) {
  const { container, getByText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText };
}
describe('Test cases for Location', () => {
  test('Create the snapshots for Location', () => {
    const { container } = renderWithContext(<Location />);

    expect(container).toMatchSnapshot();
  });

  test('Should render the title and logo correctly', () => {
    const { container } = renderWithContext(<Location />);
    const titleText = screen.getByText('Location');
    const titleComponent = container.getElementsByClassName(
      'ant-card-head-title',
    );

    expect(titleText).toBeInTheDocument();
    expect(titleText).toHaveStyle('font-size: 24px');
    expect(titleComponent[0].textContent).toContain('LocationIcon.svg');
    expect(titleComponent[0].textContent).toContain('Inclusion');
  });

  test('Should work the expand and collapse correctly when click the down arrow', () => {
    const { container } = renderWithContext(<Location />);
    const images = screen.getAllByRole('img');
    const caretDown = images[0];
    fireEvent.click(caretDown);
    const africaTestCityText = container.getElementsByClassName(
      'ant-tree-treenode-switcher-close',
    );
    expect(africaTestCityText[0].textContent).toContain('Africa');
    expect(africaTestCityText[0].textContent).not.toContain('Africa test city');
  });

  test('Should work the check and uncheck correctly when click the checkbox', () => {
    const { container } = renderWithContext(<Location />);
    const checkBoxes = container.getElementsByClassName(
      'ant-tree-checkbox-inner',
    );
    fireEvent.click(checkBoxes[1]);
    const checkedElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-checked',
    );

    const titleElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-indeterminate',
    );
    expect(checkedElement[0].textContent).toEqual('Africa test city - 1');
    expect(titleElement[0].textContent).toEqual('Africa (9,573)');

    fireEvent.click(checkBoxes[1]);
    const uncheckedElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open',
    );
    expect(uncheckedElement[0].textContent).toEqual('Africa (9,573)');
    expect(uncheckedElement[1].textContent).toEqual('Africa test city - 1');
  });

  test('Should checked all child elements when click parent element checkbox', () => {
    const { container } = renderWithContext(<Location />);
    const checkBoxes = container.getElementsByClassName(
      'ant-tree-checkbox-inner',
    );
    const parentCheckBox = checkBoxes[0];
    fireEvent.click(parentCheckBox);

    const checkedElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-checked',
    );

    const titleElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-checked',
    );

    expect(checkedElement.length).toEqual(4);
    expect(titleElement[0].textContent).toEqual('Africa (9,573)');

    fireEvent.click(checkBoxes[1]);

    const indeterminatedTitleElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-indeterminate',
    );

    expect(indeterminatedTitleElement[0].textContent).toEqual('Africa (9,573)');
  });

  test('Should checked the parent element when checked the all child elements', () => {
    const { container } = renderWithContext(<Location />);
    const checkBoxes = container.getElementsByClassName(
      'ant-tree-checkbox-inner',
    );
    const child1 = checkBoxes[1];
    const child2 = checkBoxes[2];
    const child3 = checkBoxes[3];

    fireEvent.click(child1);
    fireEvent.click(child2);
    fireEvent.click(child3);

    const checkedElement = container.getElementsByClassName(
      'ant-tree-treenode ant-tree-treenode-switcher-open ant-tree-treenode-checkbox-checked',
    );
    expect(checkedElement.length).toEqual(4);
    expect(checkedElement[0].textContent).toEqual('Africa (9,573)');
    expect(checkedElement[1].textContent).toEqual('Africa test city - 1');
    expect(checkedElement[2].textContent).toEqual('Africa test city - 2');
    expect(checkedElement[3].textContent).toEqual('Africa test city - 3');
  });

  // now it's not neccessary, in future we can use it

  //   test('Check onSelect function when click the element', () => {
  //     const { container } = renderWithContext(<Location />);

  //     const africaTestCity1 = screen.getByTitle('Africa test city - 1');

  //     fireEvent.click(africaTestCity1);

  //     const selectedElement = container.getElementsByClassName(
  //       'ant-tree-node-content-wrapper ant-tree-node-content-wrapper-normal ant-tree-node-selected',
  //     );

  //     expect(selectedElement[0].textContent).toBe('Africa test city - 1');
  //   });
});
