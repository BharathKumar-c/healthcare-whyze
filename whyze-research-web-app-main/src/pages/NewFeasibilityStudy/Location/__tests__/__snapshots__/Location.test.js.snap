// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test cases for Location Create the snapshots for Location 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-21"
    >
      <div
        class="ant-card ant-card-bordered location-Card"
        id="card_hoc"
      >
        <div
          class="ant-card-head"
        >
          <div
            class="ant-card-head-wrapper"
          >
            <div
              class="ant-card-head-title"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-center"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 20px;"
                >
                  <div
                    class="ant-space ant-space-horizontal ant-space-align-center"
                  >
                    <div
                      class="ant-space-item"
                      style="margin-right: 20px;"
                    >
                      <div>
                        <svg>
                          LocationIcon.svg
                        </svg>
                      </div>
                    </div>
                    <div
                      class="ant-space-item"
                    >
                      <div
                        style="font-weight: 700; font-size: 24px;"
                      >
                        Location
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="ant-space-item"
                >
                  <span
                    class="ant-tag titlecomponenet-tag"
                  >
                    Inclusion
                  </span>
                </div>
              </div>
            </div>
            <div
              class="ant-card-extra"
            >
              <div
                class="ant-space ant-space-horizontal ant-space-align-end"
              >
                <div
                  class="ant-space-item"
                  style="margin-right: 20px;"
                >
                  <div
                    class="card-component__Vector-icon"
                  />
                </div>
                <div
                  class="ant-space-item"
                >
                  <svg
                    class="ant-dropdown-trigger"
                    style="cursor: pointer;"
                  >
                    options.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="location-container"
          >
            <div>
              <h3>
                Top countries
              </h3>
            </div>
            <div
              class="location-container-country-tree"
            >
              <h3>
                Countries
              </h3>
              <div
                class="ant-tree"
                role="tree"
              >
                <div>
                  <input
                    aria-label="for screen reader"
                    style="width: 0px; height: 0px; display: flex; overflow: hidden; opacity: 0; border: 0px; padding: 0px; margin: 0px;"
                    tabindex="0"
                    value=""
                  />
                </div>
                <div
                  aria-hidden="true"
                  class="ant-tree-treenode"
                  style="position: absolute; pointer-events: none; visibility: hidden; height: 0px; overflow: hidden; border: 0px; padding: 0px;"
                >
                  <div
                    class="ant-tree-indent"
                  >
                    <div
                      class="ant-tree-indent-unit"
                    />
                  </div>
                </div>
                <div
                  class="ant-tree-list"
                  style="position: relative;"
                >
                  <div
                    class="ant-tree-list-holder"
                  >
                    <div>
                      <div
                        class="ant-tree-list-holder-inner"
                        style="display: flex; flex-direction: column;"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
